baseapp/abci.go

```diff
@@ -262,6 +262,27 @@ func (app *BaseApp) Commit() (res abci.ResponseCommit) {
 		app.halt()
 	}
 
+	if app.snapshotInterval > 0 && uint64(header.Height)%app.snapshotInterval == 0 {
+		go func() {
+			app.logger.Info("Taking state snapshot", "height", header.Height)
+			err := app.snapshot(uint64(header.Height))
+			if err != nil {
+				app.logger.Error("Failed to take snapshot", "height", header.Height, "err", err.Error())
+				return
+			}
+			app.logger.Info("Finished taking state snapshot", "height", header.Height)
+			if app.snapshotRetention > 0 {
+				app.logger.Info("Pruning state snapshots")
+				pruned, err := app.snapshotStore.Prune(app.snapshotRetention)
```

Comment:
we should probably have an auxiliary method to prune just like we do for snapshotting that does preliminary checks and such.

Reply:
Yeah, that makes sense.